test/unit/models/users.test.js

```diff
@@ -221,6 +221,36 @@ describe("users", function () {
     });
   });
 
+  describe("archive user if not in discord", function () {
+    beforeEach(async function () {
+      const addUsersPromises = [];
+      userDataArray.forEach((user) => {
+        const userData = {
+          ...user,
+          roles: {
+            ...user.roles,
+            in_discord: false,
+            archived: false,
+          },
+        };
+        addUsersPromises.push(userModel.add(userData));
+      });
+
+      await Promise.all(addUsersPromises);
+    });
+
+    it("should update archived role to true if in_discord is false", async function () {
+      await users.archiveUserIfNotInDiscord();
+
+      const updatedUsers = await users.fetchAllUsers();
```

Comment:
why are we fetching all users? updated users will be a small group of users than all users. fetching all users here is ambiguous.